frontend/src/pages/RequestDetailsPage/v2/KeyValueTableV2.tsx

```diff
@@ -39,3 +104,51 @@ export function KeyValueTableV2({
     </div>
   );
 }
+
+export function CollapsibleKeyValueTableV2({
+  keyValue,
+  emptyMessage = "No data",
+  className,
+  defaultCollapsed = true,
+  title,
+  sensitiveKeys = [],
+}: {
+  keyValue: Record<string, string>;
+  emptyMessage?: string;
+  className?: string;
+  defaultCollapsed?: boolean;
+  title: string;
+  sensitiveKeys?: string[];
+}) {
+  const [isOpen, setIsOpen] = useState(!defaultCollapsed);
+  const count = useMemo(() => Object.entries(keyValue).length, [keyValue]);
+  const toggleIsOpen = useCallback(() => setIsOpen((o) => !o), []);
```

Comment:
Why is this memoized? It's probably introducing more overhead than performance gains for these light functions/expressions.

Reply:
ah right, will unmemo